From: linshengjie <linshengjie@nd.com.cn>
Date: Tue, 6 Dec 2022 11:06:54 +0800
Subject: [PATCH] AP9-10703:Bouncy Balls: Balls do not jump based on sound
 level in the room fix version = 1.2.1

why:getByteFrequencyData method was annotated by patchAudioBuffer-AnalyserNode-fp-mitigations.patch, thus when it is called, an error occurs.
how:disannotate methods that are annotated by the patch.
---
 .../blink/renderer/modules/webaudio/analyser_node.idl     | 8 ++++----
 .../blink/renderer/modules/webaudio/audio_buffer.idl      | 6 ++++++
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/third_party/blink/renderer/modules/webaudio/analyser_node.idl b/third_party/blink/renderer/modules/webaudio/analyser_node.idl
--- a/third_party/blink/renderer/modules/webaudio/analyser_node.idl
+++ b/third_party/blink/renderer/modules/webaudio/analyser_node.idl
@@ -40,10 +40,10 @@

     // Copies the current frequency data into the passed array.
     // If the array has fewer elements than the frequencyBinCount, the excess elements will be dropped.
-    //[HighEntropy, Measure] void getFloatFrequencyData(Float32Array array);
-    //[HighEntropy, Measure] void getByteFrequencyData(Uint8Array array);
+    [HighEntropy, Measure] void getFloatFrequencyData(Float32Array array);
+    [HighEntropy, Measure] void getByteFrequencyData(Uint8Array array);

     // Real-time waveform data
-    //[HighEntropy, Measure] void getFloatTimeDomainData(Float32Array array);
-    //[HighEntropy, Measure] void getByteTimeDomainData(Uint8Array array);
+    [HighEntropy, Measure] void getFloatTimeDomainData(Float32Array array);
+    [HighEntropy, Measure] void getByteTimeDomainData(Uint8Array array);
 };
diff --git a/third_party/blink/renderer/modules/webaudio/audio_buffer.idl b/third_party/blink/renderer/modules/webaudio/audio_buffer.idl
--- a/third_party/blink/renderer/modules/webaudio/audio_buffer.idl
+++ b/third_party/blink/renderer/modules/webaudio/audio_buffer.idl
@@ -37,6 +37,12 @@

     // Channel access
     readonly attribute unsigned long numberOfChannels;
+    [HighEntropy=Direct, Measure, RaisesException] Float32Array getChannelData(
+        unsigned long channelIndex);
+    [HighEntropy, Measure, RaisesException] void copyFromChannel(
+        Float32Array destination,
+        unsigned long channelNumber,
+        optional unsigned long bufferOffset = 0);
     [RaisesException] void copyToChannel(
         Float32Array source,
         unsigned long channelNumber,
-- 
2.34.1

